ipatests/test_integration/test_epn.py

```diff
@@ -59,6 +62,79 @@ def postconf(host, option):
     host.run_command(r"postconf -e '%s'" % option)
 
 
+def generate_user(host, uid, **kwargs):
+    user_defaults = {
+        'objectClass': [
+            'ipaobject',
+            'person',
+            'top',
+            'ipasshuser',
+            'inetorgperson',
+            'organizationalperson',
+            'krbticketpolicyaux',
+            'krbprincipalaux',
+            'inetuser',
+            'posixaccount',
+            'ipaSshGroupOfPubKeys',
+            'mepOriginEntry'],
+        'ipaUniqueID': ['autogenerate'],
+        'loginShell': ['/bin/zsh'],
+        'uidNumber': ['-1'],
+        'gidNumber': ['-1'],
+    }
+
+    user = dict(user_defaults)
+    user['dn'] = 'uid={uid},cn=users,cn=accounts,{suffix}'.format(
+        uid=uid,
+        suffix=host.hostname,
+    )
+    user['uid'] = [uid]
+    user['displayName'] = ['{0} {0}'.format(uid)]
+    user['initials'] = ['{}{}'.format(uid[1], uid[-1])]
+    user['gecos'] = user['displayName']
+    user['sn'] = [uid]
+    user['homeDirectory'] = ['/other-home/{}'.format(uid)]
+    user['mail'] = ['{uid}@{domain}'.format(
+        uid=uid, domain=host.domain)]
+    user['krbPrincipalName'] = ['{uid}@{realm}'.format(
+        uid=uid, realm=host.realm)]
+    user['krbCanonicalName'] = ['{uid}@{realm}'.format(
+        uid=uid, realm=host.realm)]
+    user['givenName'] = [uid]
+    user['cn'] = ['{0} {0}'.format(uid)]
+
+    if "krbPasswordExpiration" in kwargs.keys():
+        user['krbPasswordExpiration'] = kwargs['krbPasswordExpiration']
+
+    if "userPassword" in kwargs.keys():
+        user['userPassword'] = kwargs['userPassword']
+
+    if "Mail2" in kwargs.keys():
+        user['Mail2'] = kwargs['Mail2']
+
+    if "Mail3" in kwargs.keys():
+        user['Mail3'] = kwargs['Mail3']
+
+    return user
+
+
+def generate_user_ldif(user_dict):
+    return ldif.CreateLDIF(dn=user_dict['dn'],record=user_dict)
```

Comment:
CreateLDIF is deprecated. Can you use ldif.LDIFWriter() instead?

Also this requires bytes, not strings, so it will fail. At least in Fedora.

Reply:
Great point